--- origsrc/claws-mail-3.13.0/configure.ac	2015-11-16 22:08:53.732031900 -0600
+++ src/claws-mail-3.13.0/configure.ac	2015-11-16 23:12:29.241540100 -0600
@@ -1163,7 +1163,7 @@ if test x"$HAVE_PERL" = xyes; then
 	PERL_LDFLAGS=`perl -MExtUtils::Embed -e ldopts |sed 's/-lgdbm//'`
 	PERL_LDFLAGS=`echo $PERL_LDFLAGS |sed 's/-ldb//'`
 	PERL_LDFLAGS=`echo $PERL_LDFLAGS |sed 's/-lndbm//'`
-	PERL_LDFLAGS=`echo $PERL_LDFLAGS |sed 's/-lc//'`
+	PERL_LDFLAGS=`echo $PERL_LDFLAGS |sed 's/-fstack-protector[^ ]*/-lssp/'`
 	AC_MSG_RESULT(ok)
 	AC_MSG_NOTICE([Adding perl LIBS ${PERL_CFLAGS}])
 	
--- origsrc/claws-mail-3.13.0/src/plugins/perl/perl_plugin.c	2015-10-11 03:19:33.000000000 -0500
+++ src/claws-mail-3.13.0/src/plugins/perl/perl_plugin.c	2015-11-16 22:57:53.785371300 -0600
@@ -520,7 +520,7 @@ static void free_all_lists(void)
 /* Initialization */
 
 /* ClawsMail::C::filter_init(int) */
-static XS(XS_ClawsMail_filter_init)
+XS(XS_ClawsMail_filter_init)
 {
   int flag;
   /* flags:
@@ -648,7 +648,7 @@ static XS(XS_ClawsMail_filter_init)
 }
 
 /* ClawsMail::C::open_mail_file */
-static XS(XS_ClawsMail_open_mail_file)
+XS(XS_ClawsMail_open_mail_file)
 {
   char *file;
   gchar buf[BUFFSIZE];
@@ -671,7 +671,7 @@ static XS(XS_ClawsMail_open_mail_file)
 }
 
 /* ClawsMail::C::close_mail_file */
-static XS(XS_ClawsMail_close_mail_file)
+XS(XS_ClawsMail_close_mail_file)
 {
   dXSARGS;
   if(items != 0) {
@@ -684,7 +684,7 @@ static XS(XS_ClawsMail_close_mail_file)
 }
 
 /* ClawsMail::C::get_next_header */
-static XS(XS_ClawsMail_get_next_header)
+XS(XS_ClawsMail_get_next_header)
 {
   gchar buf[BUFFSIZE];
   Header *header;
@@ -717,7 +717,7 @@ static XS(XS_ClawsMail_get_next_header)
 }
 
 /* ClawsMail::C::get_next_body_line */
-static XS(XS_ClawsMail_get_next_body_line)
+XS(XS_ClawsMail_get_next_body_line)
 {
   gchar buf[BUFFSIZE];
 
@@ -740,7 +740,7 @@ static XS(XS_ClawsMail_get_next_body_lin
 /* Filter matchers */
 
 /* ClawsMail::C::check_flag(int) */
-static XS(XS_ClawsMail_check_flag)
+XS(XS_ClawsMail_check_flag)
 {
   int flag;
   /* flags:  1 marked
@@ -824,7 +824,7 @@ static XS(XS_ClawsMail_check_flag)
 }
 
 /* ClawsMail::C::colorlabel(int) */
-static XS(XS_ClawsMail_colorlabel)
+XS(XS_ClawsMail_colorlabel)
 {
   int color;
 
@@ -844,7 +844,7 @@ static XS(XS_ClawsMail_colorlabel)
 }
 
 /* ClawsMail::C::age_greater(int) */
-static XS(XS_ClawsMail_age_greater)
+XS(XS_ClawsMail_age_greater)
 {
   int age;
   time_t t;
@@ -865,7 +865,7 @@ static XS(XS_ClawsMail_age_greater)
 }
 
 /* ClawsMail::C::age_lower(int) */
-static XS(XS_ClawsMail_age_lower)
+XS(XS_ClawsMail_age_lower)
 {
   int age;
   time_t t;
@@ -886,7 +886,7 @@ static XS(XS_ClawsMail_age_lower)
 }
 
 /* ClawsMail::C::tagged() */
-static XS(XS_ClawsMail_tagged)
+XS(XS_ClawsMail_tagged)
 {
   dXSARGS;
   if(items != 0) {
@@ -898,7 +898,7 @@ static XS(XS_ClawsMail_tagged)
 }
 
 /* ClawsMail::C::get_tags() */
-static XS(XS_ClawsMail_get_tags)
+XS(XS_ClawsMail_get_tags)
 {
   guint iTag;
   guint num_tags;
@@ -926,7 +926,7 @@ static XS(XS_ClawsMail_get_tags)
 
 
 /* ClawsMail::C::set_tag(char*) */
-static XS(XS_ClawsMail_set_tag)
+XS(XS_ClawsMail_set_tag)
 {
   gchar *tag_str;
   gint tag_id;
@@ -950,7 +950,7 @@ static XS(XS_ClawsMail_set_tag)
 }
 
 /* ClawsMail::C::unset_tag(char*) */
-static XS(XS_ClawsMail_unset_tag)
+XS(XS_ClawsMail_unset_tag)
 {
   gchar *tag_str;
   gint tag_id;
@@ -974,7 +974,7 @@ static XS(XS_ClawsMail_unset_tag)
 }
 
 /* ClawsMail::C::clear_tags() */
-static XS(XS_ClawsMail_clear_tags)
+XS(XS_ClawsMail_clear_tags)
 {
   dXSARGS;
   if(items != 0) {
@@ -988,7 +988,7 @@ static XS(XS_ClawsMail_clear_tags)
 
 
 /* ClawsMail::C::make_sure_tag_exists(char*) */
-static XS(XS_ClawsMail_make_sure_tag_exists)
+XS(XS_ClawsMail_make_sure_tag_exists)
 {
   gchar *tag_str;
 
@@ -1013,7 +1013,7 @@ static XS(XS_ClawsMail_make_sure_tag_exi
 
 
 /* ClawsMail::C::make_sure_folder_exists(char*) */
-static XS(XS_ClawsMail_make_sure_folder_exists)
+XS(XS_ClawsMail_make_sure_folder_exists)
 {
   gchar *identifier;
   FolderItem *item;
@@ -1034,7 +1034,7 @@ static XS(XS_ClawsMail_make_sure_folder_
 
 
 /* ClawsMail::C::addr_in_addressbook(char* [, char*]) */
-static XS(XS_ClawsMail_addr_in_addressbook)
+XS(XS_ClawsMail_addr_in_addressbook)
 {
   gchar *addr;
   gchar *bookname;
@@ -1068,7 +1068,7 @@ static XS(XS_ClawsMail_addr_in_addressbo
 /* Filter actions */
 
 /* ClawsMail::C::set_flag(int) */
-static XS(XS_ClawsMail_set_flag)
+XS(XS_ClawsMail_set_flag)
 {
   int flag;
   /* flags:  1 mark
@@ -1106,7 +1106,7 @@ static XS(XS_ClawsMail_set_flag)
 }
 
 /* ClawsMail::C::unset_flag(int) */
-static XS(XS_ClawsMail_unset_flag)
+XS(XS_ClawsMail_unset_flag)
 {
   int flag;
   /*
@@ -1145,7 +1145,7 @@ static XS(XS_ClawsMail_unset_flag)
 }
 
 /* ClawsMail::C::move(char*) */
-static XS(XS_ClawsMail_move)
+XS(XS_ClawsMail_move)
 {
   gchar *targetfolder;
   gchar *logtext;
@@ -1177,7 +1177,7 @@ static XS(XS_ClawsMail_move)
 }
 
 /* ClawsMail::C::copy(char*) */
-static XS(XS_ClawsMail_copy)
+XS(XS_ClawsMail_copy)
 {
   char *targetfolder;
   gchar *logtext;
@@ -1207,7 +1207,7 @@ static XS(XS_ClawsMail_copy)
 }
 
 /* ClawsMail::C::delete */
-static XS(XS_ClawsMail_delete)
+XS(XS_ClawsMail_delete)
 {
   dXSARGS;
   if(items != 0) {
@@ -1221,7 +1221,7 @@ static XS(XS_ClawsMail_delete)
 }
 
 /* ClawsMail::C::hide */
-static XS(XS_ClawsMail_hide)
+XS(XS_ClawsMail_hide)
 {
   dXSARGS;
   if(items != 0) {
@@ -1235,7 +1235,7 @@ static XS(XS_ClawsMail_hide)
 
 
 /* ClawsMail::C::color(int) */
-static XS(XS_ClawsMail_color)
+XS(XS_ClawsMail_color)
 {
   int color;
   gchar *logtext;
@@ -1258,7 +1258,7 @@ static XS(XS_ClawsMail_color)
 }
 
 /* ClawsMail::C::change_score(int) */
-static XS(XS_ClawsMail_change_score)
+XS(XS_ClawsMail_change_score)
 {
   int score;
   gchar *logtext;
@@ -1279,7 +1279,7 @@ static XS(XS_ClawsMail_change_score)
 }
 
 /* ClawsMail::C::set_score(int) */
-static XS(XS_ClawsMail_set_score)
+XS(XS_ClawsMail_set_score)
 {
   int score;
   gchar *logtext;
@@ -1300,7 +1300,7 @@ static XS(XS_ClawsMail_set_score)
 }
 
 /* ClawsMail::C::forward(int,int,char*) */
-static XS(XS_ClawsMail_forward)
+XS(XS_ClawsMail_forward)
 {
   int flag;
   /* flags:  1 forward
@@ -1347,7 +1347,7 @@ static XS(XS_ClawsMail_forward)
 }
 
 /* ClawsMail::C::redirect(int,char*) */
-static XS(XS_ClawsMail_redirect)
+XS(XS_ClawsMail_redirect)
 {
   int account_id,val;
   char *dest;
@@ -1391,7 +1391,7 @@ static XS(XS_ClawsMail_redirect)
 /* Utilities */
 
 /* ClawsMail::C::move_to_trash */
-static XS(XS_ClawsMail_move_to_trash)
+XS(XS_ClawsMail_move_to_trash)
 {
   FolderItem *dest_folder;
   
@@ -1415,7 +1415,7 @@ static XS(XS_ClawsMail_move_to_trash)
 }
 
 /* ClawsMail::C::abort */
-static XS(XS_ClawsMail_abort)
+XS(XS_ClawsMail_abort)
 {
   FolderItem *inbox_folder;
 
@@ -1444,7 +1444,7 @@ static XS(XS_ClawsMail_abort)
 }
 
 /* ClawsMail::C::get_attribute_value(char*,char*[,char*]) */
-static XS(XS_ClawsMail_get_attribute_value)
+XS(XS_ClawsMail_get_attribute_value)
 {
   char *addr;
   char *attr;
@@ -1472,7 +1472,7 @@ static XS(XS_ClawsMail_get_attribute_val
 }
 
 /* ClawsMail::C::filter_log(char*,char*) */
-static XS(XS_ClawsMail_filter_log)
+XS(XS_ClawsMail_filter_log)
 {
   char *text;
   char *type;
@@ -1498,7 +1498,7 @@ static XS(XS_ClawsMail_filter_log)
 }
 
 /* ClawsMail::C::filter_log_verbosity(int) */
-static XS(XS_ClawsMail_filter_log_verbosity)
+XS(XS_ClawsMail_filter_log_verbosity)
 {
   int retval;
 
