--- origsrc/claws-mail-3.9.0/src/common/ssl.c	2012-09-23 03:36:47.000000000 -0500
+++ src/claws-mail-3.9.0/src/common/ssl.c	2013-01-02 19:16:34.693872100 -0600
@@ -103,6 +103,7 @@ static int gnutls_client_cert_cb(gnutls_
 const gchar *claws_ssl_get_cert_file(void)
 {
 	const char *cert_files[]={
+		"/usr/ssl/certs/ca-bundle.crt",
 		"/etc/pki/tls/certs/ca-bundle.crt",
 		"/etc/certs/ca-bundle.crt",
 		"/etc/ssl/ca-bundle.pem",
@@ -132,6 +133,7 @@ const gchar *claws_ssl_get_cert_file(voi
 const gchar *claws_ssl_get_cert_dir(void)
 {
 	const char *cert_dirs[]={
+		"/usr/ssl/certs",
 		"/etc/pki/tls/certs",
 		"/etc/certs",
 		"/usr/share/ssl/certs",
