--- origsrc/claws-mail-3.9.2/configure.ac	2013-10-03 17:56:24.607872400 -0500
+++ src/claws-mail-3.9.2/configure.ac	2013-10-03 16:46:34.115189900 -0500
@@ -21,7 +21,7 @@ if test \( -d .git \); then
 	echo "echo ${GIT_VERSION}" > ./version
     fi
 else
-    GIT_VERSION=`sh -c ". ./$srcdir/version"`
+    GIT_VERSION=`sh -c ". $srcdir/version"`
 fi
 MAJOR_VERSION=${GIT_VERSION%%.*}
 MINOR_VERSION=${GIT_VERSION#*.}
