--- origsrc/claws-mail-3.9.2/src/common/passcrypt.c	2013-03-19 01:31:04.000000000 -0500
+++ src/claws-mail-3.9.2/src/common/passcrypt.c	2013-10-30 17:44:30.753874600 -0500
@@ -32,6 +32,8 @@
 
 #if defined (__FreeBSD__)
 #include <rpc/des_crypt.h>
+#else
+#include <crypt.h>
 #endif

 #include <glib.h>
--- origsrc/claws-mail-3.9.2/src/plugins/vcalendar/libical/libical/icalderivedproperty.c.in	2013-03-19 01:31:04.000000000 -0500
+++ src/claws-mail-3.9.2/src/plugins/vcalendar/libical/libical/icalderivedproperty.c.in	2013-10-30 18:58:33.353976800 -0500
@@ -47,6 +47,8 @@
 
 struct icalproperty_impl*
 icalproperty_new_impl (icalproperty_kind kind);
+extern void
+icalproperty_add_parameters (struct icalproperty_impl *, va_list);
 
 /* This map associates the property kinds with the string
    representation of the property name and the kind of VALUE that the

